Model/Api/ShippingMethods.php

```diff
@@ -378,49 +378,47 @@ function ($el) {
                 $prefetchAddressFields
             );
 
-            $shippingAddress = $quote->getShippingAddress();
+            $address = $quote->isVirtual() ? $quote->getBillingAddress() : $quote->getShippingAddress();
+
             // get the value of each valid field and include it in the cache identifier
             foreach ($prefetchAddressFields as $key) {
                 $getter = 'get'.$key;
-                $value = $shippingAddress->$getter();
+                $value = $address->$getter();
                 if ($value) $cache_identifier .= '_'.$value;
             }
 
             $cache_identifier = md5($cache_identifier);
 
             if ($serialized = $this->cache->load($cache_identifier)) {
-                $shippingAddress->setShippingMethod(null)->save();
+                $address->setShippingMethod(null)->save();
                 return unserialize($serialized);
             }
         }
         ////////////////////////////////////////////////////////////////////////////////////////
 
         // Get region id
-        $region = $this->regionModel->loadByName(@$shipping_address['region'], @$shipping_address['country_code']);
+        $region = $this->regionModel->loadByName(@$address_data['region'], @$address_data['country_code']);
 
         // Check the email address
-        $email = $this->cartHelper->validateEmail(@$shipping_address['email']) ? $shipping_address['email'] : null;
+        $email = $this->cartHelper->validateEmail(@$address_data['email']) ? $address_data['email'] : null;
 
-        $shipping_address = [
-            'country_id' => @$shipping_address['country_code'],
-            'postcode'   => @$shipping_address['postal_code'],
-            'region'     => @$shipping_address['region'],
+        // Reformat address data
+        $address_data = [
+            'country_id' => @$address_data['country_code'],
+            'postcode'   => @$address_data['postal_code'],
+            'region'     => @$address_data['region'],
             'region_id'  => $region ? $region->getId() : null,
-            'firstname'  => @$shipping_address['first_name'],
-            'lastname'   => @$shipping_address['last_name'],
-            'street'     => trim(@$shipping_address['street_address1'] . "\n" . @$shipping_address['street_address2']),
-            'city'       => @$shipping_address['locality'],
-            'telephone'  => @$shipping_address['phone'],
+            'street'     => trim(@$address_data['street_address1'] . "\n" . @$address_data['street_address2']),
```

Comment:
I think city may be needed for shipping calculation?

Reply:
It is not needed, but it was there. I probably deleted it by mistake. It's back.